R/methods-Vision.R

```diff
@@ -115,6 +117,14 @@ setMethod("Vision", signature(data = "matrixORSparse"),
                     )
                 data <- data[ !(rownames(data) %in% toRemove), , drop = FALSE]
             }
+
+            if (!is.null(tree)) {
+                # Subset matrix
+                data <- data[,tree$tip.label]
+                # Subset Tree
+                tree <- keep.tip(tree, colnames(data)) 
```

Comment:
Actually the best way to do this is the following:

```
cellbcs = intersect(colnames(data), tree$tip.label)
data <- data[,cellbcs]
tree <- keep.tip(tree, cellbcs)
```

This is because if there exists tip labels that do not appear in the data matrix, there will be an error.